release/tar/linux/opensearch-tar-build.sh

```diff
@@ -93,6 +94,13 @@ then
     exit 1
 fi
 
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
+then
+    echo "error: The underlying architecture($UNDERLYING_ARCH) doesnot match the product architecture($ARCHITECTURE) that you are trying to build"
+    echo "Please use the appropriate build environment"
```

Comment:
`appropriate` doesn't tell someone that sees this failure how they can resolve it, could we be specific about the kind of environment that should be setup, e.g. `To build arm64 you need to be on an arm64 architecture environment?`

Reply:
Done